doc/configuration.md

```diff
@@ -787,6 +787,32 @@ With the configuration excerpt:
 
 Then the `Phar::getMetadata()` will return `['application_version' => '1.0.0-dev']` array.
 
+If you specify a php file, the return contents (as array) will be included as metadata.
+
+For example, if you take the following code:
+
+```php
+<?php
+# filename:  /project-root/box-metadata.php
+class MyBox
+{
+    public static function generateMetadata()
```

Comment:
What are your thoughts on adding a docblock here to describe when someone would use this functionality?

```suggestion
    /**
     * In this scenario, we're providing a function to generate the
     * PHAR metadata because it's complex, better expressed as a function,
     * or is sourced from dynamic data.
     *
     * @return array
     */
    public static function generateMetadata()
```